### PR TITLE
Remove the `dat` folders to save littlefs space

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
This PR removes the `dat` folder used by `ClientFilestore` that wastes a lot of space on the filesystem.

Depends on https://github.com/Nitrokey/trussed/pull/34

This PR also adds a `migrate` function that allows moving from the previous layout to the new layout.

The migrate function cannot by itself know if the migration has been done. I would probably have it done by the Admin App, through a "privilegded" syscall, since the admin app is loaded first and can keep track of whether the migration has already been done.

### CODE REVIEW

**Comment by sosthene-nitrokey** [comment]
I made the migration and the removal of the dat folder optional, so this can be merged without staying a pain having to constantly rebase.

File: Cargo.toml [hunk 1]
```diff
@@ -20,12 +20,20 @@ serde-byte-array = "0.1.2"
 sha2 = { version = "0.10.6", default-features = false }
 subtle = { version = "2.4.1", default-features = false }
 trussed = { version = "0.1.0", features = ["serde-extensions"] }
+littlefs2 = "0.4.0"
+admin-app = "0.1.0"
```
Comment (robin-nitrokey) [inline Cargo.toml:24]: unneeded dependency?

**Review by robin-nitrokey — COMMENTED** [review]
I feel like the `use_raw` configuration option is leaking implementation details.  And without context, it is not clear which setting is the “correct”/new one.

What do you think about just using a versioning system? That would make it easy to identify the correct setting, and the implementation details would stay in this crate.  Potentially it could also be used to select the correct migration steps.

```rust
pub enum FilesystemLayout {
    V0,
    V1,
}

impl FilesystemLayout {
    fn use_raw(&self) -> bool {
        self != FilesystemLayout::V0
    }
}
```

File: Cargo.toml (hunk 1 above)
Comment (sosthene-nitrokey) [reply Cargo.toml:24]: Right admin-app can `dev-dependency` only.

**Comment by sosthene-nitrokey** [comment]
I like that. Will do this. For se050 backend too.

**Review by robin-nitrokey — APPROVED** [review]
Thank you!

**Pull request merged**